src/main/scala/PrivateHome/data.scala

```diff
@@ -1,47 +1,180 @@
+    withSQL {
+      insertInto(Device).values(device.id, "", device.switchtype, if (device.keepStatus) device.Status else 0, device.keepStatus)
+    }.update.apply
+    device match {
+      case device: mhzSwitch => withSQL {
+        insertInto(Mhz).values(device.id, device.systemCode, device.unitCode)
+      }.update().apply()
+        mhzId += ((device.systemCode + device.unitCode, device.id))
+      case _: mqttSwitch =>
+      case _ => throw wrongclass
+    }
+    devices = devices.concat(Map((device.id, device)))
   }
+  //ToDo: add an settings XML
+  //ToDo: add support for Settingschanges
 
-  val settings: Map[String, Int] = Map(("port", 2888), ("ip", 565))
-
-  def idTest(id:String):Unit ={
+  def idTest(id: String): Unit = {
     if (id.length != 5) throw new IllegalArgumentException("""Length of ID is not 5""")
     if (!id.matches("[-_a-zA-Z0-9]{5}")) throw new IllegalArgumentException("""ID Contains not Allowed Characters""")
-    if (data.IDs.contains(id))throw new IllegalArgumentException("""ID is already used""")
+    if (devices.contains(id)) throw new IllegalArgumentException("""ID is already used""")
+  }
+
+  /**
+   * An better access to the devices Map
+   *
+   * @param deviceID The ID of the switch you want
+   * @return the Switch object the you requested
+   */
+  def getDevice(deviceID: String): Switch = {
+    devices(deviceID)
+  }
+
+  /**
+   * Message class for devices Table
+   *
+   * @param id         ID of the Device in the Format [0-9a-Z] five character long
+   * @param name       The name of the Device any String lenght in the Table 64 character
+   * @param switchtype A String identification of the Switch type max length 16 character
+   * @param state      an foatingpoint representation of the State when keepState is true 4 decimalpoints long
+   * @param keepState  Boolean that indicates if the State should be restored at turn on
+   */
+  case class Device(id: String, name: String, switchtype: String, state: Float, keepState: Boolean)
+
+  /**
+   * Message class for Mhz Table only needed when Switchtype is MQTT
+   *
+   * @param id         ID of the Device in the Format [0-9a-Z] five character long
+   * @param systemcode Systemcode for the Mhzreciever
+   * @param unitcode   Unitcode for the Mhzreciever
+   */
+  case class Mhz(id: String, systemcode: String, unitcode: String)
+
+  /**
+   * Syntaxsupport Object for devices table
+   */
+  object Device extends SQLSyntaxSupport[Device] {
+    override val tableName = "devices"
+
+    def apply(rs: WrappedResultSet) = new Device(
+      rs.string("id"), rs.string("name"), rs.string("type"), rs.float("state"), rs.boolean("keepState"))
+  }
+
+  /**
+   * Syntaxsupport Object for Mhz table
+   */
+  object Mhz extends SQLSyntaxSupport[Mhz] {
+    override val tableName = "Mhz"
+
+    def apply(rs: WrappedResultSet): Mhz = new Mhz(rs.string("id"), rs.string("systemcode"), rs.string("unitcode"))
   }
 
```

Comment:
MQTT!!!

Reply:
NO this is the syntax support for the Mhz table in the database no need for a MQTT one

Reply:
ok